contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,398 @@
+
+![A/B test set up](../images/tutorials/redirect-testing/test.png)
+
+Because we are working locally, you can click "Launch" immediately.
+
+## Setting up the redirect test middleware
+
+Next.js enables you to run [middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) that intercepts and modifies requests for your app. We use it for our redirect test.
+
+To start, create a `middleware.js` file in the base `redirect-test` directory. We want it to run on both the `/test` and `/control` paths, so we add them to the matcher config. For now, we have the `/test` path redirect to `/control` as a placeholder.
+
+```js
+// redirect-test/middleware.js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  if (request.nextUrl.pathname=== '/test') {
+   // Placeholder for now. We'll replace this code in the next step
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  return NextResponse.next()
+}
+
+export const config = {
+  matcher: ['/test', '/control'],
+};
+```
+
+### Getting or creating a user ID for flag evaluation
+
+To evaluate the experiment flag in our middleware, we need a distinct user ID to target. This needs to be consistent because we want the same user to see the same variant of the experiment every time.
+
+To do this, we need to:
+
+1. Check if a `distinct_id` exists in the PostHog cookie, and use it if so.
+2. Create a `distinct_id` if not.
+
+This requires using your project API key to get the cookies, parsing them as JSON, and potentially creating a distinct ID using `crypto.randomUUID()`. Altogether, this looks like this:
+
+```js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  const ph_project_api_key = '<ph_project_api_key>'
+  const ph_cookie_key = `ph_${ph_project_api_key}_posthog`
+  const cookie = request.cookies.get(ph_cookie_key);
+
+  let distinct_id;
+  if (cookie) {
+    // Use PostHog distinct_id 
+    distinct_id = JSON.parse(cookie.value).distinct_id;
+  } else {
+    // Create new distinct_id
+    distinct_id = crypto.randomUUID();
+  }
+//... rest of code
+```
+
+### Evaluating our redirect test with PostHog
+
+With our distinct ID, we use the PostHog API to check the value of the `main-redirect` feature flag for a user (because [we can’t use PostHog SDKs in Next.js middleware](https://vercel.com/docs/functions/edge-functions/edge-runtime#supported-apis)). This is known as evaluating the feature flag.
+
+Specifically, we evaluate the flag by making a POST request to the `https://app.posthog.com/decide?v=3` route (replace `app` with `eu` if you’re on EU Cloud) with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, it looks like this:
```

Comment:
Since we mention the eu endpoint in the code comment, I dont think we need the explanation
```suggestion
Specifically, we evaluate the flag by making a POST request to the `[https://app.posthog.com/decide?v=3](/docs/api/post-only-endpoints#feature-flags)` route with your project API key and user distinct ID. From the response, we get the value of the `main-redirect` feature flag and use it to redirect to the right page. Altogether, it looks like this:
```